File: examples/snippets/modules/ROOT/examples/build-akka-streamlets-java/step0/src/main/java/com/example/ReportPrinter.java
package com.example;

import akka.NotUsed;
import akka.stream.*;
import akka.stream.javadsl.*;

import cloudflow.streamlets.*;
import cloudflow.streamlets.avro.*;
import cloudflow.akkastream.*;
import cloudflow.akkastream.javadsl.*;

public class ReportPrinter extends AkkaStreamlet {
  // 1. TODO Create inlets and outlets
  // 2. TODO Define the shape of the streamlet
  public StreamletShape shape() { throw new UnsupportedOperationException("Not Implemented"); }
  // 3. TODO Override createLogic to provide StreamletLogic
  public RunnableGraphStreamletLogic createLogic() { throw new UnsupportedOperationException("Not Implemented"); }
}

File: examples/snippets/modules/ROOT/examples/build-akka-streamlets-scala/step0/src/main/scala/com/example/ReportPrinter.scala
package com.example

import cloudflow.akkastream._
import cloudflow.streamlets.StreamletShape

//TODO rename to ReportPrinter
object ReportPrinterStep0 extends AkkaStreamlet {
  // 1. TODO Create inlets and outlets
  // 2. TODO Define the shape of the streamlet
  override val shape: StreamletShape = StreamletShape.empty
  // 3. TODO Override createLogic to provide StreamletLogic
  override def createLogic: AkkaStreamletLogic = new AkkaStreamletLogic() { override def run: Unit = () }
}
